File: app/src/main/java/com/holokenmod/DigitSetting.java
package com.holokenmod;

import java.util.ArrayList;
import java.util.Collection;

public enum DigitSetting {
    FIRST_DIGIT_ONE,
    FIRST_DIGIT_ZERO;


    public Collection<Integer> getPossibleDigits(final int gridSize) {
        final Collection<Integer> digits = new ArrayList<>();

        for (int i = getMinimumDigit(); i <= getMaximumDigit(gridSize); i++) {
            digits.add(i);
        }

        return digits;
    }

    public int getMaximumDigit(final int gridSize) {
        if (this == FIRST_DIGIT_ONE) {
            return gridSize;
        }

        return gridSize - 1;
    }

    public int getMinimumDigit() {
        if (this == FIRST_DIGIT_ONE) {
            return 1;
        }

        return 0;
    }

    public Collection<Integer> getPossibleNonZeroDigits(final int gridSize) {
        final Collection<Integer> digits = new ArrayList<>();

        for (int i = 1; i <= getMaximumDigit(gridSize); i++) {
            digits.add(i);
        }

        return digits;
    }
}

File: gauguin-core/src/main/kotlin/org/piepmeyer/gauguin/options/DigitSetting.kt
package org.piepmeyer.gauguin.options

import org.piepmeyer.gauguin.grid.GridSize

enum class DigitSetting(
    val numbers: Set<Int>,
) {
    FIRST_DIGIT_ONE((1..12).toSet()),
    FIRST_DIGIT_ZERO((0..11).toSet()),
    PRIME_NUMBERS(setOf(1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31)),
    FIBONACCI_SEQUENCE(
        setOf(
            1,
            2,
            3,
            5,
            8,
            13,
            21,
            34,
            55,
            89,
            144,
            233,
        ),
    ),
    PADOVAN_SEQUENCE(
        setOf(1, 2, 3, 4, 5, 7, 9, 12, 16, 21, 28, 37),
    ),
    FIRST_DIGIT_MINUS_TWO(
        (-2..9).toSet(),
    ),
    FIRST_DIGIT_MINUS_FIVE((-5..6).toSet()),
    ;

    fun getPossibleDigits(gridSize: GridSize): Set<Int> = numbers.take(gridSize.amountOfNumbers).toSet()

    fun getMaximumDigit(gridSize: GridSize): Int = numbers.elementAt(gridSize.amountOfNumbers - 1)

    fun getPossibleNonZeroDigits(gridSize: GridSize): Set<Int> =
        getPossibleDigits(gridSize)
            .filterNot { it == 0 }
            .toSet()

    fun indexOf(value: Int): Int = numbers.indexOf(value)

    fun zeroOnKeyPadShouldBePlacedAtLast(): Boolean = numbers.first() == 0
}
